### DISCUSSION TITLE
Multihop selects different countries for entry and exit

### DISCUSSION THREAD

**Discussion by acrostich** [mullvad/mullvadvpn-app#9104] [Feature requests & Ideas]
### I have checked if others have suggested this already

- [X] I have checked this issue tracker to see if others have reported similar issues.

### Feature description

When using a (large) list of multiple countries for both entry and exit locations, the client often selects the same country for both entry and exit, which defeats some reasons to use the multihop features. Choose different countries for entry and exit locations when several are included in the list used for both.

One way to achieve this would be to (randomly) choose the entry point server then (randomly) choose the exit point in a subset of the list without the servers in the country choosen for the entry point.

### Alternative solutions

The current workaround is to create different non-overlapping lists for entry and exit points, but this restriction the choice and is further complicated when one wants to use DAITA as one will also have to track when new servers/countries support this feature.

### Type of feature

- [X] Better privacy/anonymity
- [ ] Better at circumventing censorship
- [X] Easier to use
- [ ] Other

### Operating System

- [X] Android
- [X] iOS
- [X] Windows
- [X] macOS
- [X] Linux

**Comment by Serock3** [comment]
Thank you for this feature request! We are planning to rework the location select view, and this could something to consider at that stage. I'll leave this issue open to track the engagement.

To reduce confusion. let me clarify how we select relays for multihop.
* Without DAITA, we select the entry and exit relays randomly from the list of those that match your chosen locations and other constraints. It may or may not be the same country, there is no preference. If you have selected two countries with different amount of servers, we are more likely to pick a server from the country with the most amount of servers.
* With DAITA, unless you enable "Direct Only", we pick the entry relay for you and always choose the one closest to the exit. This will likely end up being in the same country.
